### PR TITLE
[v6r21]Add remove ReconnectListener in disconnect

### PR DESCRIPTION
**Pull request by wkrzemien**
As suggested by Chris this removes the infinite loop due to
disconnect/reconnect sequences.

BEGINRELEASENOTES

*Resources/MessageQueue
FIX: Remove infinite loop bug caused by the ReconnectListener.
ENDRELEASENOTES

### CODE REVIEW

**Comment by chaen** [comment]
Ping ? :)

**Comment by fstagni** [comment]
Hi @wkrzemien we'd need this PR to be ready for merging ASAP (basically today)

**Comment by wkrzemien** [comment]
@fstagni @chaen  I added and extended the integration tests related to producers and the last problems. I have no idea why travis shows some errors that seems totally unrelated to my code changes. I will try again later.
I also modified the MonitoringReporter code to reuse the same default producer and not recreating new over and over @zmathe can you please have a look?
If you think that those changes in MonitoringReporter are not necessary I can revert it back.

**Comment by fstagni** [comment]
The Travis issues are for the latest version of M2Crypto which is automatically picked up (and does not work), see also https://gitlab.com/m2crypto/m2crypto/issues/255

For the rest the PR looks OK to me.

**Comment by fstagni** [comment]
Just merge the head of rel-v6r21 in your branch and Travis should be fine.

**Comment by wkrzemien** [comment]
@fstagni  - done

File: MonitoringSystem/Client/MonitoringReporter.py [hunk 1]
```diff
@@ -152,12 +155,19 @@ def commit(self):
...
-      if mqProducer is not None:
-        mqProducer.close()
-        mqProducer = None
       self.__documents.extend(documents)
     return S_OK(recordSent)
 
+  def __getProducer(self):
+    """
+    This method is used to get the default MQ producer or create it if needed.
+    Returns:
```
Comment (andresailer) [inline MonitoringSystem/Client/MonitoringReporter.py]: Needs a new line before

Comment (wkrzemien) [reply MonitoringSystem/Client/MonitoringReporter.py]: Is it ok now?

Comment (andresailer) [reply MonitoringSystem/Client/MonitoringReporter.py]: Yes, thanks.

**Comment by fstagni** [comment]
Weirdly Travis run the tests against the integration branch, so what's reported for the format is wrong. @chaen please check the content of this one.

**Comment by chaen** [comment]
The change for the MQ seems fine to me, but @zmathe said he will look at the MonitorignReporter part

**Comment by zmathe** [comment]
I do not see any problem with the reporter. If the reconnection works fine, we do not have to recreate always the producer and consumer.

**Pull request merged**